### PR TITLE
Clean up documentation, move changelog to GitHub releases

### PR DESCRIPTION
**Pull request by NextTurn**
- Mainly cleans up the formatting.
- Updates links.
- Refactors the CHANGELOG to follow the Release Drafter format and moves it to Releases.

Please review the CHANGELOG first. You can preview the changes in Releases.

Note that relative links in Releases are all permanent links,
e.g. https://github.com/winsw/winsw/blob/v2.1.0/doc/xmlConfigFile.md#delayedautostart

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,217 +1,107 @@
-Release Notes
-====
+# Release Notes
 
-##### Newer releases
-
-See [GitHub Releases](https://github.com/winsw/winsw/releases)
+See [Releases](https://github.com/winsw/winsw/releases) and [Milestones](https://github.com/winsw/winsw/milestones).
```
Comment (oleg-nenashev) [inline CHANGELOG.md]: The milestones link might be confusing unless documented what it means

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,217 +1,107 @@
...
-Add a new `hidewindow` option to suppress windows popup on legacy platforms 
-* [PR #330](https://github.com/kohsuke/winsw/pull/330) -
-Fix duplicate startup log entries
-* [PR #299](https://github.com/kohsuke/winsw/pull/299),
-  [PR #303](https://github.com/kohsuke/winsw/pull/303),
-  [PR #311](https://github.com/kohsuke/winsw/pull/311),
-  [PR #316](https://github.com/kohsuke/winsw/pull/316),
-  [PR #318](https://github.com/kohsuke/winsw/pull/318) -
-  Documentation cleanup
+* Prevent application crash when `roll-by-size-time` is enabled (#315)
```
Comment (oleg-nenashev) [inline CHANGELOG.md]: Markdown pages do not resolve simple issue/PR references even when viewed from the GitHub WebUI

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,217 +1,107 @@
...
+* Add a new `hidewindow` option to suppress windows popup on legacy platforms (#321)
+* Fix duplicate startup log entries (#330)
+* Documentation cleanup (#299, #303, #311, #316, #318)
 
-Internal changes:
+## Internal changes
```
Comment (oleg-nenashev) [inline CHANGELOG.md]: Wrong header levels here and below. Version header is h5 while this one is h2

File: doc/extensions/extensions.md [hunk 4]
```diff
@@ -1,7 +1,6 @@
-WinSW extensions
-===
+# WinSW extensions
 
-Starting from WinSW `2.0`, the wrapper provides an internal extension engine and several extensions.
+Starting from WinSW v2, the wrapper provides an internal extension engine and several extensions.
```
Comment (oleg-nenashev) [inline doc/extensions/extensions.md]: why? It was a specific version link

File: doc/installation.md [hunk 5]
```diff
@@ -68,50 +67,50 @@ In particular, the following option can be set up:
...
+  then *myapp.exe* will launch the executable specified in the configuration file (Java in this case). 
+  If this process dies, *myapp.exe* will exit itself, and the service will be considered stopped.
+
 ### Extra configuration options
-  
-#### Making WinSW 1.x compatible with .NET runtime 4.0+
 
-**NOTE.** _Starting from WinSW `2.0` the release offers a new binary, which targets the .NET Framework 4.0.
-Such configuration is no longer required._  
+#### Making WinSW v1 compatible with .NET runtime 4.0+
```
Comment (oleg-nenashev) [inline doc/installation.md]: Should just archive it later

File: doc/xmlConfigFile.md [hunk 6]
```diff
@@ -1,27 +1,26 @@
-WinSW XML Configuration File
-====
+# WinSW XML Configuration File
 
-This page describes the configuration file, which controls the behavior of the Jenkins service.
+This page describes the configuration file, which controls the behavior of the Windows service.
```
Comment (oleg-nenashev) [inline doc/xmlConfigFile.md]: thanks for catching

File: doc/xmlConfigFile.md [hunk 7]
```diff
@@ -170,40 +180,42 @@ For servers requiring authentication some parameters must be specified depending
...
+For target servers using the HTTPS transfer protocol it is necessary, that the CA which issued the server certificate is trusted by the client. This is normally the situation when the server ist located in the Internet. When an organisation is using a self issued CA for the intranet this probably is not the case. In this case it is necessary to import the CA to the Certificate MMC of the Windows client. Have a look to the instructions on this [site](https://technet.microsoft.com/en-us/library/cc754841.aspx). The self issued CA must be imported to the Trusted Root Certification Authorities for the computer.
```
Comment (oleg-nenashev) [inline doc/xmlConfigFile.md]: For the future we need to splitting docs to one sentence-per-line

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,217 +1,107 @@
-Release Notes
-====
+# Release Notes
```
Comment (oleg-nenashev) [inline CHANGELOG.md]:
```suggestion
# Release Notes (Archive)
```

**Comment by NextTurn** [comment]
Editing existing releases doesn't raise notifications. I have already copied the change logs to releases entirely. Could you look over there first?

**Comment by oleg-nenashev** [comment]
> Editing existing releases doesn't raise notifications. I have already copied the change logs to releases entirely. Could you look over there first?

Oh right. It is not a problem for Existing releases, only for newly created ones if there is no GitHub release associated with a tag. Sorry for confusion.

**Comment by oleg-nenashev** [comment]
So yes, then I am fine with the change, except the header levels which need to be fixed. I can do it on the top of your pull request if you are fine with it

File: CHANGELOG.md (hunk 3 above)
Comment (NextTurn) [reply CHANGELOG.md]: So I'm going to delete these since they're now present in Releases.

Comment (oleg-nenashev) [reply CHANGELOG.md]: Fine with me

File: doc/extensions/extensions.md (hunk 4 above)
Comment (NextTurn) [reply doc/extensions/extensions.md:3]: Reverted.

**Comment by NextTurn** [comment]
Most headers in the Markdown files should be promoted by one level.

**Pull request merged**